app/models/storage.rb

```diff
@@ -71,9 +71,9 @@ class Storage < ApplicationRecord
   SUPPORTED_STORAGE_TYPES = %w( VMFS NFS NFS41 FCP ISCSI GLUSTERFS )
 
   supports :smartstate_analysis do
-    if ext_management_systems.blank? || !ext_management_system.class.supports_smartstate_analysis?
```

Comment:
I think a blank ems_id is a case we should still handle, storages didn't used to be cascade deleted with their ext_management_system so there are plenty of archived ones